packages/lesswrong/components/tagging/AllWikiTagsPage.tsx

```diff
@@ -341,99 +258,73 @@ const ArbitalRedirectNotice = ({ classes, onDismiss }: {
 
 const AllWikiTagsPage = () => {
   const classes = useStyles(styles);
-  const { openDialog } = useDialog();
-  const currentUser = useCurrentUser();
 
-  const { SectionButton, WikiTagNestedList, LWTooltip } = Components;
+  const { WikiTagGroup, Loading, SectionButton, LWTooltip } = Components;
 
   const { query } = useLocation();
   const isArbitalRedirect = query.ref === 'arbital';
 
-  const { data } = useQuery(allTagsQuery, { ssr: true });
-  const tags: AllTagsPageCacheFragment[] = data?.AllTags ?? [];
-
-  // State variable for the current search query
-  const [currentQuery, setCurrentQuery] = useState('');
-  const [showArbitalRedirectNotice, setShowArbitalRedirectNotice] = useState(isArbitalRedirect);
-
-  // Function to handle search state changes
-  const handleSearchStateChange = (searchState: any): void => {
-    setCurrentQuery(searchState.query);
-  };
-
-  const priorityTagIds = filterNonnull(
-    prioritySlugs.map(slug => tags.find(tag => isMatchingCoreTag(tag, slug)))
-  ).map(tag => tag._id);
-
-  const adjustedItems = tags.map(tag => {
-    if (priorityTagIds.includes(tag._id as typeof priorityTagIds[number])) {
-      return {
-        ...tag,
-        parentTagId: null,
-      };
-    }
+  const currentUser = useCurrentUser();
+  const { openDialog } = useDialog();
 
-    return {
-      ...tag,
-      parentTagId: tag.coreTagId ? tag.coreTagId : 'uncategorized-root',
-    };
+  const { results: priorityTagsRaw } = useMulti({
+    collectionName: "Tags",
+    fragmentName: "ConceptItemFragment",
+    terms: { 
+      view: "tagsBySlugs",
+      slugs: prioritySlugs as unknown as string[] 
```

Comment:
If it complained about the type being a `readonly ...`, you should just be able to do `slugs: [...prioritySlugs]` instead